prime-router/src/main/kotlin/common/HttpClientUtils.kt

```diff
@@ -0,0 +1,582 @@
+        }
+
+        /**
+         * Common helper for external func
+         */
+        private fun invoke(
+            method: HttpMethod,
+            url: String,
+            tokens: BearerTokens? = null,
+            headers: Map<String, String>? = null,
+            acceptedContent: ContentType? = ContentType.Application.Json,
+            timeout: Long? = REQUEST_TIMEOUT_MILLIS,
+            queryParameters: Map<String, String>? = null,
+            jsonPayload: String? = null,
+            httpClient: HttpClient? = null,
+        ): HttpResponse {
+            return runBlocking {
+                (httpClient ?: createDefaultHttpClient(tokens)).request(url) {
+                    this.method = method
+                    timeout {
+                        requestTimeoutMillis = timeout
+                    }
+                    url {
+                        queryParameters?.forEach {
+                            parameter(it.key, it.value.toString())
+                        }
+                    }
+
+                    headers?.let {
+                        headers {
+                            headers.forEach {
+                                append(it.key, it.value)
+                            }
+                        }
+                    }
+                    acceptedContent?.let {
+                        accept(acceptedContent)
+                        contentType(acceptedContent)
+                    }
+                    jsonPayload?.let {
+                        setBody(jsonPayload)
+                    }
+                }
+            }
+        }
+
+        /**
+         * Create a http client with sensible default settings
+         * note: most configuration parameters are overridable
+         * e.g. expectSuccess default to false because most of the time
+         * the caller wants to handle the whole range of response status
+         * @param bearerTokens null default, the access token needed to call the endpoint
+         * @return a HttpClient with all sensible defaults
+         */
+        fun createDefaultHttpClient(bearerTokens: BearerTokens?): HttpClient {
+            return HttpClient(Apache) {
+                // installs logging into the call to post to the server
+                // commented out - not to override underlying default logger settings
+                // enable to trace http client internals when needed
+                // install(Logging) {
+                //     logger = Logger.SIMPLE
+                //     level = LogLevel.INFO
+                // }
+                bearerTokens?.let {
+                    install(Auth) {
+                        bearer {
+                            loadTokens {
+                                bearerTokens
+                            }
+                            refreshTokens {
+                                bearerTokens
+                            }
+                        }
+                    }
+                }
```

Comment:
Since this does a ton under the hood it would probably be easier to just nix it and add the header by hand.

```kotlin
bearerTokens?.accessToken?.let {
    headers {
        append("Authorization", "Bearer $it")
    }
}
```

Reply:
I think thats a premature optimization though. We currently don't use refresh tokens and it caused us an issue. I think these few lines are what we actually want to do at the moment and if we do want to use refresh tokens in the future it will be easy to find where the auth header is currently being appended.

Reply:
Yeah you're right. Copacetic :).